inst/pages/98_exercises.qmd

```diff
@@ -1146,13 +1434,141 @@ Useful functions: scater::runMDS, runNMDS, transformAssay, ggplot, scater::plotR
 
 ### Heatmap visualization
 
-1. Load experimental dataset from mia.
-2. Visualize abundances with heatmap.
-3. Visualize abundances with heatmap after CLR + Z transformation.
+1. Load one of the datasets from mia as well as miaViz.
+
+```{r}
+#| label: heatmap-ex1
+#| code-fold: true
+#| code-summary: "Show solution"
+#| eval: false
+
+library(mia)
+library(miaViz)
+data("GlobalPatterns", package = "mia")
+tse <- GlobalPatterns
+```
+
+2. Add a relative abundance assay to the TSE.
+
+```{r}
+#| label: heatmap-ex2
+#| code-fold: true
+#| code-summary: "Show solution"
+#| eval: false
+
+tse <- transformAssay(tse, assay.type = "counts", method = "relabundance")
+```
+
+3. Melt the tse into a dataframe, order it by decreasing relative abundance and subset to the first 25 rows for better visibility when plotting the heatmap.
+
+```{r}
+#| label: heatmap-ex3
+#| code-fold: true
+#| code-summary: "Show solution"
+#| eval: false
+
+df <- meltAssay(tse, assay.type = "relabundance")
+
+df <- df[order(abs(df$relabundance), decreasing=TRUE), ]
+
+top_taxa_samples <- df[1:20,1:2]
+
+df <- df[df$FeatureID %in%top_taxa_samples$FeatureID & 
+             df$SampleID %in%top_taxa_samples$SampleID, ]
+```
```

Comment:
This seems to be little bit complex for basic heatmap exercise. You could first subset the data based on prevalence or other with mia tools. Then melt the data,

Reply:
I was also wondering if we should use sechm package instead of creating these heatmaps "manually." That package creates ComplexHeatmaps directly from SummarizedExperiment objects. It lacks some nice functionality but could be still usable in basic situations.

Reply:
Right. Might be good to support SE-based solutions.